### PR TITLE
[OpaqueValues] Open opaque existential value.

### PR DESCRIPTION
**Pull request by nate-chandler**
When a value's preferred existential representation is opaque, if the value is not an address (which can only happen in opaque values mode), emit `open_existential_value` rather than `open_existential_addr`.

### CODE REVIEW

**Comment by nate-chandler** [comment]
@swift-ci please test

**Comment by nate-chandler** [comment]
@swift-ci please test linux platform

**Comment by nate-chandler** [comment]
@swift-ci please test windows platform

**Comment by nate-chandler** [comment]
@swift-ci plase test linux platform

**Comment by nate-chandler** [comment]
@swift-ci plase test windows platform

**Comment by nate-chandler** [comment]
@swift-ci plase test linux platform

File: lib/SILGen/SILGenLValue.cpp [hunk 1]
```diff
@@ -946,17 +946,31 @@ namespace {
...
+          auto borrow =
+              SGF.B.createBeginBorrow(loc, base.getValue(), /*isLexical=*/false,
+                                      /*hasPointerEscape=*/false);
+          auto value =
+              SGF.B.createOpenExistentialValue(loc, borrow, getTypeOfRValue());
+
+          SGF.Cleanups.pushCleanup<EndBorrowCleanup>(borrow);
+          return ManagedValue::forForwardedRValue(SGF, value);
+        } else {
+          addr = SGF.B.createOpenExistentialAddr(
```
Comment (slavapestov) [inline lib/SILGen/SILGenLValue.cpp:969]: Can you `assert(SGF.useLoweredAddresses());` here?

Comment (nate-chandler) [reply lib/SILGen/SILGenLValue.cpp:969]: No, we can't--even in opaque values mode we still need to open lvalues with `open_existential_addr`.  For example:
```
protocol I {
  mutating func foo()
}

func f(_ i: inout any I) {
  i.foo()
}
```

**Comment by nate-chandler** [comment]
@swift-ci please test macos platform

**Pull request merged**